instrumentation/resources/library/src/main/java/io/opentelemetry/instrumentation/resources/internal/ContainerResourceComponentProvider.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.resources.internal;
+
+import com.google.auto.service.AutoService;
+import io.opentelemetry.instrumentation.resources.ContainerResource;
+import io.opentelemetry.sdk.autoconfigure.spi.internal.ComponentProvider;
+import io.opentelemetry.sdk.autoconfigure.spi.internal.StructuredConfigProperties;
+import io.opentelemetry.sdk.resources.Resource;
+
+/**
+ * Declarative config container resource provider.
+ *
+ * <p>This class is internal and is hence not for public use. Its APIs are unstable and can change
+ * at any time.
+ */
+@SuppressWarnings("rawtypes")
+@AutoService(ComponentProvider.class)
+public class ContainerResourceComponentProvider implements ComponentProvider<Resource> {
```

Comment:
I think we could move the repeating parts to a super class 
```java
public ContainerResourceComponentProvider() {
  super(ContainerResource::get);
}

class ResourceComponentProvider implements ComponentProvider<Resource> {
  private final Supplier<Resource> supplier;
  ResourceComponentProvider(Supplier<Resource> supplier) {
    this.supplier = supplier;
  }

  @Override
  public Resource create(StructuredConfigProperties config) {
    return supplier.get();
  }
}
```
Idk whether it is worth the effort.

Reply:
I like it 👍